core/lib/spree/core/state_machines/order.rb

```diff
@@ -128,10 +128,12 @@ def define_state_machine!
               end
 
               after_transition do |order, transition|
+                Spree::Event.fire "order_transitioned_to_#{transition.to}", order: order
                 order.logger.debug "Order #{order.number} transitioned from #{transition.from} to #{transition.to} via #{transition.event}"
               end
 
               after_failure do |order, transition|
+                Spree::Event.fire "order_failed_transition_to_#{transition.to}", order: order
```

Comment:
What about `order_failed_transitioning_to_...`?

Reply:
It could also work no problem, I have no strong opinions on `transition` vs `transitioning`.

Reply:
Let's wait for the opinion of @spaghetticode, which has more experience with events' names

Reply:
Personally, I guess I am slightly partial to `transitioned_to`, as it's a bit shorter. 

In general, I think we should strive to make these event names descriptive and easily mappable to the state machine transition that triggered them. One extra bonus may be having names with consistent/predictable format, so if `after_failure` spawns the event `order_failed_transition_to_completed`, then for symmetry the name fo `after_transition` may be changed to `order_successful_transition_to_completed`. Unfortunately, it's a bit too long 🤷.

Maybe we can spend a little more time in searching the best names, also considering that changing them later will require a deprecation cycle.